### ISSUE TITLE
Wiki module version 2.x no longer displays hierarchical tree structure of Wiki when moving an article to a new category

### ISSUE THREAD

**Issue by RayCulp** [humhub/wiki#292]
### Humhub version
1.14.4

### Wiki module version
2.0.1

### Steps to reproduce
1. Edit an existing Wiki article
2. Click field Parent page

### Expected behavior
Hierarchical tree structure of Wiki is displayed in drop-down list, allowing user to choose new category from the list.

### Actual behavior
User is prompted to enter at least two characters to search for new category. A flat list of search results / matches is then displayed.

### Comments
Regrettably, this turns what is otherwise a very nicely redesigned module into a complete non-starter for us. We have Wikis with thousands of pages and complex hierarchical structures that are constantly being edited and restructured. We sincerely hope there are plans to reintroduce this feature. In the meantime, we will attempt to revert back to version 1.9.0. If this is not possible, we will bite the bullet and restore the latest backup, thereby losing a few days work in the process, but this is better than losing the hierarchical structure of categories.

### Screenshots

![image](https://github.com/humhub/wiki/assets/7621330/7c6f156d-e064-4e90-a546-0f04e44d6bcf)

**Comment by RayCulp** [comment]
Version 2.x appears to be incompatible with 1.x, so we reverted to a backup of the site.

**Comment by yurabakhtin** [comment]
@luke- Should we just revert the selector to previous version:

![wiki-parent-selector](https://github.com/humhub/wiki/assets/6995524/cc79469f-bbaf-4037-87f0-e830ccd719d5)

or should we implement some switcher/toggler button between simple select input and search selector?
It can be located under the selector or right after the header "Parent Page" and stored per current user in local storage.
Or we can implement this as module setting per container Space/User.

(it was changed here https://github.com/humhub/team_tasks/issues/259#issuecomment-1318550227)

**Comment by luke-** [comment]
@yurabakhtin Thanks for info, let's leave it as it is for now.  
@RayCulp It is also possible to sort and group pages via drag and drop on the overview page or with the Enterprise Theme on the left navigation.

**Comment by RayCulp** [comment]
@luke- Thanks for the suggestions. We do not have the Enterprise Theme, but I just tested the new version again, and it does indeed seem possible now to drag and drop categories and pages in the overview. This is something that never worked correctly for us before. Nevertheless, just out of curiosity, what exactly is the advantage of the flat, unordered list that has replaced the previous hierarchical list? There must have been a good reason to introduce it that I am not seeing yet.

**Comment by luke-** [comment]
@RayCulp I see advantages in the fast search by keyword and also performance.  But I also understand your point and we will reconsider.

**Comment by atdqm** [comment]
@luke-  @RayCulp 

`Nevertheless, just out of curiosity, what exactly is the advantage of the flat, unordered list that has replaced the previous hierarchical list?`

I also prefer the dropdown hierarchical list, because we don't have to know the name of the categories and where they are located

**Comment by RayCulp** [comment]
> I also prefer the dropdown hierarchical list, because we don't have to know the name of the categories and where they are located

I agree. In some cases, the name of the subcategory does not actually contain enough information to be able to find it without the additional information from the main category above it.

Example:

1. Main category 1: How to fry an egg
1.1  Subcategory 1.1 - Ingredients for this task

2. Main category 2: How to make pancakes
2.2 Subcategory 2.1 - Ingredients you will need

If you are moving an article from a third category or location, the names of the subcategories do not contain enough information for you to know which one is the correct one to place it in.

Thankfully, drag and drop in the overview actually works well now.
